modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/index/DynamicEnableIndexingBasicSelfTest.java

```diff
@@ -0,0 +1,168 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.ignite.Ignite;
+import org.apache.ignite.IgniteException;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.cache.CacheAtomicityMode;
+import org.apache.ignite.cache.CacheMode;
+import org.apache.ignite.cluster.ClusterState;
+import org.apache.ignite.configuration.CacheConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.configuration.NearCacheConfiguration;
+import org.apache.ignite.internal.IgniteEx;
+import org.apache.ignite.internal.util.typedef.G;
+import org.apache.ignite.testframework.GridTestUtils;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+/**
+ * Tests basic functionality of enabling indexing.
+ */
+@RunWith(Parameterized.class)
+public class DynamicEnableIndexingBasicSelfTest extends DynamicEnableIndexingAbstractTest {
+    /** Test parameters. */
+    @Parameters(name = "hasNear={0},nodeIdx={1},cacheMode={2},atomicityMode={3}")
+    public static Iterable<Object[]> params() {
+        int[] opNodes = new int[] {IDX_CLI, IDX_SRV_CRD, IDX_SRV_NON_CRD, IDX_SRV_FILTERED};
+
+        CacheMode[] cacheModes = new CacheMode[] {CacheMode.PARTITIONED, CacheMode.REPLICATED};
+
+        CacheAtomicityMode[] atomicityModes = new CacheAtomicityMode[] {
+                CacheAtomicityMode.ATOMIC,
+                CacheAtomicityMode.TRANSACTIONAL,
+                CacheAtomicityMode.TRANSACTIONAL_SNAPSHOT
+        };
+
+        List<Object[]> res = new ArrayList<>();
+
+        for (int node : opNodes) {
+            for (CacheMode cacheMode : cacheModes) {
+                for (CacheAtomicityMode atomicityMode : atomicityModes) {
+                    res.add(new Object[] {true, node, cacheMode, atomicityMode});
+
+                    // For TRANSACTIONAL_SNAPSHOT near caches is forbidden.
+                    if (atomicityMode != CacheAtomicityMode.TRANSACTIONAL_SNAPSHOT)
+                        res.add(new Object[] {false, node, cacheMode, atomicityMode});
+
+                }
+            }
+        }
+
+        return res;
+    }
+
+    /** */
+    @Parameter(0)
+    public Boolean hasNear;
+
+    /** */
+    @Parameter(1)
+    public int nodeIdx;
+
+    /** */
+    @Parameter(2)
+    public CacheMode cacheMode;
+
+    /** */
+    @Parameter(3)
+    public CacheAtomicityMode atomicityMode;
+
+    /** {@inheritDoc} */
+    @Override protected void beforeTestsStarted() throws Exception {
+        super.beforeTestsStarted();
+
+        for (IgniteConfiguration cfg : configurations())
+            Ignition.start(cfg);
+
+        node().cluster().state(ClusterState.ACTIVE);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected void afterTestsStopped() throws Exception {
+        stopAllGrids();
```

Comment:
AFAIK `stopAllGrids` automatically executed after any test. 
Why we have to perform it explicitly?